### PR TITLE
Hotfix/php81 docker drupal update

### PR DESCRIPTION
**Pull request by martinyde**
https://jira.itkdev.dk/browse/HRIN-731

- Add github actions
- Update php version (8.1)
- Update drupal core (9.5.8)
- Update docker environment (3.0.1)
- Update contrib modules (The lot - i.e. "composer update --with-all-dependencies")
- Apply coding standards
- Change backend theme to Claro

### CODE REVIEW

File: patches/deskpro/deskpro-api-client-php/addTypeHinting.patch
Comment (rimi-itk) [inline patches/deskpro/deskpro-api-client-php/addTypeHinting.patch:1]: Why is this patch added? We can update the Deskpro API module with the changes.

File: web/modules/custom/hoeringsportal_deskpro/composer.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
   },
   "require": {
     "deskpro/deskpro-api-client-php": "^2.0",
-    "drupal/advancedqueue": "^1.0",
+    "drupal/advancedqueue": "1.0.0-rc3",
```
Comment (rimi-itk) [inline web/modules/custom/hoeringsportal_deskpro/composer.json:15]: `^1.0` allows installation of `1.0.0-rc3` (as I understand it) so we should not change this.

File: web/themes/custom/hoeringsportal/README.md [hunk 2]
```diff
@@ -16,19 +16,22 @@ JavaScript and CSS (actually SCSS) assets are put in `assets/js/` and
 First, install tools and requirements:
 
 ```sh
-docker run --rm --tty --interactive --volume ${PWD}:/app node:16 yarn --cwd /app install
+docker run --rm --tty --interactive --volume ${PWD}:/app node:16 yarn --cwd 
+/app install
```
Comment (rimi-itk) [inline web/themes/custom/hoeringsportal/README.md:20]: Remove the line breaks.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: web/modules/custom/hoeringsportal_deskpro/composer.json (hunk 1 above)
Comment (martinyde) [reply web/modules/custom/hoeringsportal_deskpro/composer.json:15]: ^1.0 also allows for installation of 1.0.0-rc7 which will cause https://www.drupal.org/files/issues/2021-10-20/add-getjobs-method-database-backend.patch patch to fail. It looks like a patch we need, so until there is an available patch in the https://www.drupal.org/project/advancedqueue/issues/3244940 thread or we update our code to not use getJobs() method we are stuck at rc3

File: patches/deskpro/deskpro-api-client-php/addTypeHinting.patch
Comment (martinyde) [reply patches/deskpro/deskpro-api-client-php/addTypeHinting.patch:1]: This is a patch to https://github.com/deskpro/deskpro-api-client-php library required by hoeringsportal_deskpro module, located in vendor library.

File: web/modules/custom/hoeringsportal_deskpro/README.md [hunk 3]
```diff
@@ -25,23 +25,30 @@ drush advancedqueue:queue:process hoeringsportal_deskpro
 ## Building assets
 
 ```sh
-docker compose run --rm node yarn --cwd web/modules/custom/hoeringsportal_deskpro install
-docker compose run --rm node yarn --cwd web/modules/custom/hoeringsportal_deskpro build
+docker compose run --rm node yarn --cwd
+ web/modules/custom/hoeringsportal_deskpro install
+docker compose run --rm node yarn --cwd
+ web/modules/custom/hoeringsportal_deskpro build
```
[the remark this answers is not in the document]
Comment (martinyde) [reply web/modules/custom/hoeringsportal_deskpro/README.md]: Removed line breaks and added .markdownlintrc

File: web/themes/custom/hoeringsportal/README.md (hunk 2 above)
Comment (martinyde) [reply web/themes/custom/hoeringsportal/README.md]: Fixed

File: web/modules/custom/hoeringsportal_deskpro/composer.json (hunk 1 above)
Comment (rimi-itk) [reply web/modules/custom/hoeringsportal_deskpro/composer.json:15]: We should do something about out use of `getJobs`: https://github.com/itk-dev/hoeringsportal/blob/master/web/modules/custom/hoeringsportal_deskpro/src/Service/HearingHelper.php#L374-L378

File: patches/deskpro/deskpro-api-client-php/addTypeHinting.patch
Comment (rimi-itk) [reply patches/deskpro/deskpro-api-client-php/addTypeHinting.patch:1]: Oh, I thought `deskpro-api-client-php` was our module. Why do we need to patch a vendor module with type hints?

File: web/modules/custom/hoeringsportal_deskpro/composer.json (hunk 1 above)
Comment (martinyde) [reply web/modules/custom/hoeringsportal_deskpro/composer.json]: Moved getJobs method to hoeringsportalHelper class as a private method. Updated advanced queue module.

File: composer.json [hunk 4]
```diff
@@ -25,14 +26,20 @@
             "type": "path",
             "url": "web/modules/custom/hoeringsportal_data",
             "options": {
-                "symlink": false
+                "symlink": false,
+                "versions": {
+                    "drupal/hoeringsportal_data": "1.0-dev"
```
Comment (rimi-itk) [inline composer.json:31]: 🎉

File: composer.json [hunk 5]
```diff
@@ -58,8 +65,8 @@
         "drupal/entityqueue": "^1.0@alpha",
         "drupal/field_group": "^3.0",
         "drupal/flag": "^4.0@alpha",
-        "drupal/hoeringsportal_data": "*",
-        "drupal/hoeringsportal_deskpro": "*",
+        "drupal/hoeringsportal_data": "^1.0",
+        "drupal/hoeringsportal_deskpro": "1.0-dev",
```
Comment (rimi-itk) [inline composer.json:69]: I suppose we could require both custom modules as version `^1.0`.

File: composer.json [hunk 6]
```diff
@@ -166,7 +189,7 @@
         "enable-patching": true,
         "patches": {
           "deskpro/deskpro-api-client-php": {
-            "Add typehinting to APIResponse": "patches/deskpro/deskpro-api-client-php/addTypeHinting.patch"
+            "Add typehinting to APIResponse to avoid drupal warnings in log": "patches/deskpro/deskpro-api-client-php/addTypeHinting.patch"
```
Comment (rimi-itk) [inline composer.json:192]: 👍

File: web/modules/custom/hoeringsportal_deskpro/src/Service/HearingHelper.php [hunk 7]
```diff
@@ -371,7 +371,7 @@ public function synchronizeTicket(array $payload, $delayed = TRUE) {
 
       if ($backend instanceof SupportsDeletingJobsInterface) {
         /** @var \Drupal\advancedqueue\Job $job */
-        foreach ($backend->getJobs(Job::STATE_QUEUED) as $job) {
+        foreach ($this->getJobs(Job::STATE_QUEUED) as $job) {
```
Comment (rimi-itk) [inline web/modules/custom/hoeringsportal_deskpro/src/Service/HearingHelper.php:374]: 👍 🎉

File: web/modules/custom/itk_admin/src/Form/AdminSettingsForm.php [hunk 8]
```diff
@@ -41,7 +41,7 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     $links = '';
     foreach ($tasks['tabs'] as $tab) {
       $link_array = Link::createFromRoute($tab['#link']['title'], $tab['#link']['url']->getRouteName())->toRenderable();
-      $links .= '<div>' . render($link_array) . '</div>';
+      $links .= '<div>' . \Drupal::service('renderer')->render($link_array) . '</div>';
```
Comment (rimi-itk) [inline web/modules/custom/itk_admin/src/Form/AdminSettingsForm.php:44]: We should use dependency injection to get the `renderer` service.

File: composer.json [hunk 9]
```diff
@@ -107,21 +117,34 @@
...
+          "phpstan analyse --configuration=phpstan.neon"
+        ],
+        "code-analysis": [
+          "@code-analysis/phpstan"
+        ],
         "check-coding-standards/phpcs": [
             "vendor/bin/phpcs --standard=phpcs.xml.dist"
         ],
-        "check-coding-standards": [
+        "coding-standards-check": [
```
Comment (rimi-itk) [inline composer.json:134]: For consistency we should change “check-coding-standards” to “coding-standards-check” (“apply-coding-standards” to “coding-standards-apply”) throughout this file.

**Review by rimi-itk — APPROVED** [review]
Approved with a few comments.

**Pull request merged**